api/v1/lib/httpcli/httpsched/httpsched.go

```diff
@@ -173,15 +184,31 @@ func (cli *client) httpDo(ctx context.Context, m encoding.Marshaler, opt ...http
 	opt = append(opt, httpcli.Context(ctx))
 	for attempt := 0; ; attempt++ {
 		resp, err = cli.Client.Do(m, opt...)
-		redirectErr, ok := err.(*mesosRedirectionError)
-		if !ok {
-			return resp, err
+		if err == nil {
+			return
 		}
+		redirectErr, ok := err.(*mesosRedirectionError)
+
 		if attempt < cli.redirect.MaxAttempts {
+			var candidate string
+			if !ok {
+				if cli.candidateSelector == nil {
+					log.Printf("not found candidate selector, using url when initilize framework")
+					candidate = cli.Endpoint()
+				} else {
+					candidate = cli.candidateSelector()
+				}
+				if candidate == "" {
+					log.Printf("not found candidate url, return directly")
+					return
+				}
+			} else {
+				candidate = redirectErr.newURL
```

Comment:
maybe make this the default value of `candidate` when it's initially declared and get rid of this `else` statement

Reply:
get newURL from redirectErr object is only safe when the ok == null. (unless there will be some nil point error)
this else block seems is necessary at here